// aws-cdk/lib/commands/flags/flags.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlagCommandHandler = void 0;
const interactive_handler_1 = require("./interactive-handler");
const obsolete_flags_1 = require("./obsolete-flags");
const operations_1 = require("./operations");
const router_1 = require("./router");
const validator_1 = require("./validator");
class FlagCommandHandler {
    /** Main component that sets up all flag operation components */
    constructor(flagData, ioHelper, options, toolkit, cliContextValues = {}) {
        this.flags = flagData.filter(flag => !obsolete_flags_1.OBSOLETE_FLAGS.includes(flag.name));
        this.options = { ...options, concurrency: options.concurrency ?? 4 };
        this.ioHelper = ioHelper;
        const validator = new validator_1.FlagValidator(ioHelper);
        const flagOperations = new operations_1.FlagOperations(this.flags, toolkit, ioHelper, cliContextValues);
        const interactiveHandler = new interactive_handler_1.InteractiveHandler(this.flags, flagOperations);
        this.router = new router_1.FlagOperationRouter(validator, interactiveHandler, flagOperations);
    }
    /** Main entry point that processes the flags command */
    async processFlagsCommand() {
        if (this.flags.length === 0) {
            await this.ioHelper.defaults.error('The \'cdk flags\' command is not compatible with the AWS CDK library used by your application. Please upgrade to 2.212.0 or above.');
            return;
        }
        await this.router.route(this.options);
    }
}
exports.FlagCommandHandler = FlagCommandHandler;
